File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
def solution(k, tangerine):
    dict1 = {}
    for i in tangerine:
        if i in dict1:
            dict1[i] += 1
        else:
            dict1[i] = 1
    sorted_dict = sorted(dict1.items(), key= lambda x: x[1], reverse=True)
    answer = 0
    for j in sorted_dict:
        if k <= 0:
            break
        else:
            k -= j[1]
            answer += 1
    return answer

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.js
function solution(k, tangerine) {
    let answer = 0;
    let map = new Map();
    for(let i=0; i<tangerine.length; i++){
        if(map.get(tangerine[i]) == undefined){
            map.set(tangerine[i],1);
        }
        else{
            map.set(tangerine[i],map.get(tangerine[i]) +1);
        }
    }
    let mapToArray = Array.from(map);
    mapToArray.sort((a,b) => b[1] - a[1]);
    for(let i=0; i<mapToArray.length; i++){
        if(mapToArray[i][1] >= k){
            answer += 1;
            break;
        }
        else{
            k -= mapToArray[i][1];
            answer += 1;
        }
    }
    return answer;
}